src/command_modules/azure-cli-extension/azure/cli/command_modules/extension/tests/latest/test_extension_commands.py

```diff
@@ -185,6 +185,41 @@ def test_list_available_extensions_custom_index_url(self):
             list_available_extensions(index_url=index_url)
             c.assert_called_once_with(index_url)
 
+    def test_add_list_show_remove_extension_extra_index_url(self):
+        """
+        Tests extension addition while specifying --extra-index-url parameter.
+        :return:
+        """
+        test_index = 'https://testpypi.python.org/simple'
+
+        add_extension(source=MY_EXT_SOURCE, extra_index_urls=[test_index])
+        actual = list_extensions()
+        self.assertEqual(len(actual), 1)
+        ext = show_extension(MY_EXT_NAME)
+        self.assertEqual(ext[OUT_KEY_NAME], MY_EXT_NAME)
+        remove_extension(MY_EXT_NAME)
+        num_exts = len(list_extensions())
+        self.assertEqual(num_exts, 0)
+
+    def test_update_extension_extra_index_url(self):
+        """
+        Tests extension update while specifying --extra-index-url parameter.
+        :return:
+        """
+        test_index = 'https://testpypi.python.org/simple'
+        add_extension(source=MY_EXT_SOURCE, extra_index_urls=[test_index])
+        ext = show_extension(MY_EXT_NAME)
+        self.assertEqual(ext[OUT_KEY_VERSION], '0.0.3+dev')
+        newer_extension = _get_test_data_file('myfirstcliextension-0.0.4+dev-py2.py3-none-any.whl')
+        computed_extension_sha256 = _compute_file_hash(newer_extension)
+        with mock.patch('azure.cli.command_modules.extension.custom.resolve_from_index', return_value=(newer_extension, computed_extension_sha256)):
+            update_extension(MY_EXT_NAME, extra_index_urls=[test_index])
+        ext = show_extension(MY_EXT_NAME)
+        self.assertEqual(ext[OUT_KEY_VERSION], '0.0.4+dev')
+        # Removing extension for cleanup of this test.
+        remove_extension(MY_EXT_NAME)
+        pass
```

Comment:
Clean up will be done in `tearDown` above so that is not needed.
Also, `pass` isn't needed.